File: 739-daily-temperatures/daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        wait_days = [0] * len(temperatures)
        st = []
        for i in range(len(temperatures)):
            while st and temperatures[i] > temperatures[st[-1]]:
                wait_days[st[-1]] = i - st[-1]
                st.pop()
            st.append(i)
        return wait_days

File: 739-daily-temperatures/daily-temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        vector<int> ans(temperatures.size());
        stack<int> st;
        for (int i = 0; i < temperatures.size(); i++) {
            while (!st.empty() && temperatures[i] > temperatures[st.top()]) {
                int popped = st.top();
                ans[popped] = i - popped;
                st.pop();
            }
            st.push(i);
        }
        return ans;
    }
};